cdap-common/src/main/java/co/cask/cdap/common/lang/ClassPathResources.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.common.lang;
+
+import co.cask.cdap.common.internal.guava.ClassPath;
+import co.cask.cdap.common.internal.guava.ClassPath.ResourceInfo;
+import com.google.common.base.Function;
+import com.google.common.base.Splitter;
+import com.google.common.base.Throwables;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Sets;
+import org.apache.twill.api.ClassAcceptor;
+import org.apache.twill.internal.utils.Dependencies;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Collection;
+import java.util.Set;
+
+/**
+ * Utility methods for {@link ClassPath} {@link ResourceInfo resources}.
+ */
+public final class ClassPathResources {
```

Comment:
Yes, ``ProgramResources`` had all this functionality, but was specific for ``Program``s. I extracted out the common functionality into ``ClassPathResources``, so both ``Program``s (via ``ProgramResources``) and extensions can use it.